app/controllers/verify/address_controller.rb

```diff
@@ -7,10 +7,9 @@ class AddressController < ApplicationController
     def index; end
 
     def create
-      response = Idv::AddressDeliveryMethodForm.new.submit(
-        address_delivery_params.to_h.symbolize_keys
-      )
-
+      address_delivery_hash = address_delivery_params.to_h
+      analytics.track_event(Analytics::ADDRESS_DELIVERY_METHOD, address_delivery_hash)
```

Comment:
For consistency with our controller/form object pattern, what do you think about these small tweaks:
```ruby
result = Idv::AddressDeliveryMethodForm.new.submit(address_delivery_hash.symbolize_keys)

analytics.track_event(Analytics::ADDRESS_DELIVERY_METHOD, result.to_h)
```

And then in `Idv::AddressDeliveryMethodForm`:
```ruby
def submit(address_delivery_method: '')
  self.address_delivery_method = address_delivery_method

  FormResponse.new(success: valid_address_delivery_method?, errors: {}, extra: extra_analytics_attributes)
end

private

def extra_analytics_attributes
  {
    address_delivery_method: address_delivery_method,
  }
end
```

Reply:
Some more tweaks:
I don't think we should be converting the strong parameters to a hash. Can we keep this consistent with all our other controllers?
```ruby
result = Idv::AddressDeliveryMethodForm.new.submit(address_delivery_params)
```
And in `Idv::AddressDeliveryMethodForm`, modify the submit method to take in the strong params as an argument:
```ruby
def submit(params)
  self.address_delivery_method = params[:address_delivery_method]

  FormResponse.new(success: valid_address_delivery_method?, errors: {}, extra: extra_analytics_attributes)
end
```

Reply:
It seems less clear and more error prone to be logging the result of the submit to the analytics for that specific event.  Originally I was even going to pull out the actual parameter I needed and log only that.  Moreover, if the form submit explodes for any reason you lose the analytics which is why I did it before the call.  That said this form submit won't explode and if we're clear about what the submit returns I'm ok with logging that to the analytics especially if it make the code more uniform.